cpp/validate_harvested_records.cc

```diff
@@ -229,30 +220,43 @@ int Main(int argc, char *argv[]) {
     unsigned total_record_count(0), new_record_count(0), missed_expectation_count(0);
     while (const auto record = reader->read()) {
         ++total_record_count;
-        const auto journal_name(GetJournalNameOrDie(record));
-
-        auto journal_name_and_info(journal_name_to_info_map.find(journal_name));
-        bool first_record(false); // True if the current record is the first encounter of a journal
-        if (journal_name_and_info == journal_name_to_info_map.end()) {
-            first_record = true;
-            JournalInfo new_journal_info;
-            LoadFromDatabaseOrCreateFromScratch(&db_connection, journal_name, &new_journal_info);
-            journal_name_to_info_map[journal_name] = new_journal_info;
-            journal_name_and_info = journal_name_to_info_map.find(journal_name);
-        }
+        bool validation_failed(false);
+
+        // Intentionally true to allow early breaking
+        while (true) {
+            const auto journal_name(record.getSuperiorTitle());
+            if (journal_name.empty()) {
+                LOG_WARNING("Record w/ control number \"" + record.getControlNumber() + "\" is missing a superior title!");
```

Comment:
Wouldn't a LOG_ERROR be reasonable here because it would be some logic error if we attempted to harvest an article for a journal for which we have no title?